test/org/sagebionetworks/bridge/models/schedules/CriteriaScheduleStrategyTest.java

```diff
@@ -0,0 +1,348 @@
+        
+        // Group not prohibited so first schedule returned
+        Schedule schedule = getScheduleFromStrategy(Sets.newHashSet("groupNotProhibited"));
+        assertEquals(strategyWithOneProhibitedDataGroup, schedule);
+        
+        // this group is prohibited so second schedule is returned
+        schedule = getScheduleFromStrategy(Sets.newHashSet("group1"));
+        assertEquals(strategyNoCriteria, schedule);
+    }
+    
+    @Test
+    public void filtersOnProhibitedDataGroups() {
+        setUpStrategyWithProhibitedDataGroups();
+        setUpStrategyNoCriteria();
+        
+        // context has a prohibited group, so the last schedule is returned
+        Schedule schedule = getScheduleFromStrategy(Sets.newHashSet("group1"));
+        assertEquals(strategyNoCriteria, schedule);
+        
+        // context has one of the prohibited groups, same thing
+        schedule = getScheduleFromStrategy(Sets.newHashSet("foo","group1"));
+        assertEquals(strategyNoCriteria, schedule);
+        
+        // context has no prohibited groups, first schedule is returned
+        schedule = getScheduleFromStrategy(Sets.newHashSet());
+        assertEquals(strategyWithProhibitedDataGroups, schedule);
+    }
+    
+    @Test
+    public void noMatchingFilterReturnsNull() {
+        setUpStrategyWithAppVersions();
+        setUpStrategyWithProhibitedDataGroups();
+        
+        User user = getUser();
+        user.setDataGroups(Sets.newHashSet("group1"));
+        ScheduleContext context = new ScheduleContext.Builder()
+                .withClientInfo(ClientInfo.fromUserAgentCache("app/44"))
+                .withUser(user).build();
+
+        Schedule schedule = strategy.getScheduleForUser(plan, context);
+        assertNull(schedule);
+    }
+    
+    @Test
+    public void canMixMultipleCriteria() {
+        setUpStrategyWithAppVersions();
+        setUpStrategyWithOneRequiredDataGroup();
+        setUpStrategyWithProhibitedDataGroups();
+        
+        User user = getUser();
+        user.setDataGroups(Sets.newHashSet("group3"));
+        ScheduleContext context = new ScheduleContext.Builder()
+                .withClientInfo(ClientInfo.fromUserAgentCache("app/44"))
+                .withUser(user).build();
+        
+        // First two don't match because app version is wrong and missing required groups
+        // The last is returned because the context has no prohibited groups
+        Schedule schedule = strategy.getScheduleForUser(plan, context);
+        assertEquals(strategyWithProhibitedDataGroups, schedule);
+    }
+
+    @Test
+    public void canGetAllPossibleScheduled() {
+        setUpStrategyWithAppVersions();
+        setUpStrategyWithOneRequiredDataGroup();
+        setUpStrategyWithProhibitedDataGroups();
+
+        List<Schedule> schedules = strategy.getAllPossibleSchedules();
+        assertEquals(3, schedules.size());
+    }
+    
+    @Test
+    public void validates() {
```

Comment:
Add test cases to validates:
* criteria w/o a schedule
* criteria w/ invalid schedule
* happy case: CriteriaScheduleStrategy that passes validation